module/VuFind/src/VuFind/OAuth2/Entity/ScopeEntity.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * OAuth2 scope entity implementation.
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2022.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  OAuth2
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+namespace VuFind\OAuth2\Entity;
+
+use League\OAuth2\Server\Entities\ScopeEntityInterface;
+use League\OAuth2\Server\Entities\Traits\EntityTrait;
+use League\OAuth2\Server\Entities\Traits\ScopeTrait;
+
+/**
+ * OAuth2 scope entity implementation.
+ *
+ * @category VuFind
+ * @package  OAuth2
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class ScopeEntity implements ScopeEntityInterface
+{
+    use EntityTrait, ScopeTrait;
+
+    /**
+     * Scope description
+     *
+     * @var string
+     */
+    protected $description;
+
+    /**
+     * Whether the scope is hidden from the scope list
+     *
+     * @var bool
+     */
+    protected $hidden;
+
+    /**
+     * Whether the scope requires data from an ILS account
+     *
+     * @var bool
+     */
+    protected $ilsNeeded;
+
+    /**
+     * Constructor
+     *
+     * @param array $config Client configuration
+     */
+    public function __construct(array $config)
+    {
+        foreach (['identifier'] as $required) {
+            if (!isset($config[$required])) {
+                throw new \Exception("OAuth2 scope config missing '$required'");
+            }
+        }
+        $this->setIdentifier($config['identifier']);
+        $this->setDescription($config['description'] ?? '');
+        $this->setILSNeeded(boolval($config['ils'] ?? false));
```

Comment:
Any strong reason to use `boolval` instead of `(bool)` typecasting? Nothing wrong with it, but I feel like it's not how we usually force values to booleans in the code base.

Reply:
No, changed.